src/parsing.cpp

```diff
@@ -28,16 +29,18 @@ std::vector < std::string > parsing::domain_and_port(std::string& url){
   std::string holding;
   unsigned int output_offset = 0;
   
-  // Check for the presence of user authentication info. If it exists, dump it.
-  // Use a query-check here because some people put @ info in params, baaah
+  // Check for auth credentials
   std::size_t f_param = url.find("?");
   std::size_t auth;
-  if(f_param != std::string::npos){
-    auth = url.substr(0, f_param).find("@");
-  } else {
-    auth = url.find("@");
-  }
-  if(auth != std::string::npos){
+  std::regex rx ("^(\\w+:\\w+@).*");
```

Comment:
Might suggest compiling the regex on package instantiation for a speed bump rather than recompiling it with every individual call? This is going to mean a lot of compilations!